clippy_lints/src/types.rs

```diff
@@ -1449,3 +1450,326 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for InvalidUpcastComparisons {
         }
     }
 }
+
+/// **What it does:** Checks for `impl` or `fn` missing generalization over
+/// different hashers and implicitly defaulting to the default hashing
+/// algorithm (SipHash). This lint ignores private free-functions.
+///
+/// **Why is this bad?** `HashMap` or `HashSet` with custom hashers cannot be
+/// used with them.
+///
+/// **Known problems:** Suggestions for replacing constructors are not always
+/// accurate.
```

Comment:
In what way are they inaccurate? Something a human can trivially fix? Or might there be cases that can't be resolved at all due to constraints by external crates?

Reply:
My thought was that there can be a `HashMap` which has the same types as `Self` but not exposed to outside.

```rust
impl Foo for HashMap<K, V> {
    fn method(&self) {
        let used_internally = HashMap::<K, V>::new(); // suggestion for this can be unwanted
    }
}
```

But yes constraints by external crates are more problematic.